### PR TITLE
Add support for initializing network with different parameters

### PR DESCRIPTION
**Pull request by Ttl**
Adds support for initializing network from all support parameters (s, z, y, a, h and t). Also `f` initializer can now take `Frequency` object.

### CODE REVIEW

File: skrf/network.py [hunk 1]
```diff
@@ -1322,8 +1329,11 @@ def f(self) -> npy.ndarray:
 
     @f.setter
     def f(self, f: NumberLike) -> None:
```
Comment (jhillairet) [inline skrf/network.py:1331]:
```suggestion
    def f(self, f: Union[NumberLike, Frequency]) -> None:
```

**Review by jhillairet — APPROVED** [review]
Wow, nicely done. Just a suggestion for the typing, which should be mergeable to your PR.

**Comment by jhillairet** [comment]
@Ttl , is it OK to be merged?

**Comment by Ttl** [comment]
Yes

**Pull request merged**